src/internal/NativeEncryptor.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NativeModules } from "react-native"
+import { PowerAuthCryptogram } from "../model/PowerAuthEncryptor"
+import { PowerAuthEncryptionHttpHeader } from "../model/PowerAuthEncryptionHttpHeader"
+
+/**
+ * Data returned from encryptRequest() function.
+ */
+export interface EncryptedRequestData {
+    /**
+     * Cryptogram that must be set to request body.
+     */
+    cryptogram: PowerAuthCryptogram
+    /**
+     * HTTP header indicating the encryption.
+     */
+    header: PowerAuthEncryptionHttpHeader
+    /**
+     * Identifier of newly created native object that is capable 
+     * to decrypt the response.
+     */
+    decryptorId: string
+}
+
+/**
+ * Encryptor interface implemented in the native code.
+ */
+export interface PowerAuthEncryptorIfc {
```

Comment:
Conventional TS naming for this should be something like `IPowerAuthEncryptor`

Reply:
This is internal interface, not exposed to the application. I can address this later in #150.